### PR TITLE
chore: Tool setup

### PR DESCRIPTION
**Pull request by sulha199**
### Summary

1. Setup Prettier
2. Setup ESLint
3. Setup VSCode formatting
4. Setup Prettier that runs on pre-commit
5. add tsc in npm lint to catch Typescript's strict rules violation
6. Issue https://github.com/supertokens/dashboard/issues/14

Issue #14 

### Tested on all primary browsers for:
- Chrome
    - [x] Desktop
    - [x] Mobile
    - [x] Tablet
- Safari
    - [ ] Desktop
    - [ ] Mobile
    - [ ] Tablet
- Firefox
    - [x] Desktop
    - [x] Mobile
    - [x] Tablet
- (Optional) Tested on Safari 12 (Physical or emulator)
    - [ ] iPad
    - [ ] iPhone
- (Optional) Tested on physical mobile and tablet device for:
    - [ ] Android
    - [ ] iOS (including iPadOS)

## Documentation changes

NA

## Checklist for important updates

-   [ ] Changelog has been updated
-   [ ] Changes to the version if needed
    -   In `package.json`
    -   In `package-lock.json`
    -   In `src/version.ts`
-   [x] Had run `npm run build`
-   [x] Had installed and ran the pre-commit hook

## Remaining TODOs for this PR
NA

### CODE REVIEW

**Comment by sulha199** [comment]
Hi @nkshah2 

Could you review the rules first?
Once the rules are agreed then I will fix the conflict and merge with new changes.

**Comment by nkshah2** [comment]
> Hi @nkshah2
> 
> Could you review the rules first? Once the rules are agreed then I will fix the conflict and merge with new changes.

This isnt really a priority at the moment, we will revisit this PR after the initial release

File: .husky/pre-commit [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env sh
```
Comment (nkshah2) [inline .husky/pre-commit:1]: We dont need husky at all for this, we already have a pre commit hook right? just modify that

Comment (nkshah2) [reply .husky/pre-commit:1]: Remove the dependency on husky

File: .prettierrc [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  "bracketSpacing": true,
+  "singleQuote": false,
+  "trailingComma": "es5",
+  "arrowParens": "always",
+  "jsxBracketSameLine": true,
+  "bracketSameLine": true,
+  "singleAttributePerLine": false,
+  "endOfLine": "lf",
+  "printWidth": 120
+}
```
Comment (nkshah2) [inline .prettierrc:14]: Make sure these properties match exactly

```json
{
    "tabWidth": 4,
    "semi": true,
    "singleQuote": false,
    "printWidth": 120,
    "jsxBracketSameLine": true
}
```

Also:
- `useTabs` should be true
- `singleAttributePerLine` should be true
- `endOfLine` should be removed, let prettier handle based on OS

File: .vscode/settings.json [hunk 3]
```diff
@@ -0,0 +1,45 @@
+{
```
Comment (nkshah2) [inline .vscode/settings.json:1]: If you're committing this, you should also add the extensions you need to the settings file (prettier and eslint dont work without the extensions for example)

File: package.json [hunk 4]
```diff
@@ -58,19 +58,32 @@
...
+    "eslint-plugin-n": "^15.2.4",
+    "eslint-plugin-prettier": "^4.2.1",
+    "eslint-plugin-promise": "^6.0.0",
+    "eslint-plugin-react": "^7.30.1",
     "eslint-webpack-plugin": "^3.1.1",
     "express": "^4.17.1",
     "file-loader": "^6.2.0",
     "fs-extra": "^10.0.0",
     "helmet": "^5.1.0",
+    "husky": "^8.0.0",
```
Comment (nkshah2) [inline package.json:79]: Remove

**Review by nkshah2 — CHANGES_REQUESTED** [review]

File: package.json (hunk 4 above)
Comment (sulha199) [reply package.json:79]: husky removed

File: .prettierrc (hunk 2 above)
Comment (sulha199) [reply .prettierrc:14]: updated

File: .vscode/settings.json (hunk 3 above)
Comment (sulha199) [reply .vscode/settings.json:1]: add prettier & eslint into extensions.json

The html, json, javascript, typescript are built in feature in VScode

File: package.json [hunk 5]
```diff
@@ -84,7 +96,7 @@
     "start-bundle-server": "rm -rf debug && BROWSER=none node scripts/start.js",
     "start-dev-server": "npx ts-node-dev --project server/tsconfig.json server/index.ts",
     "build": "rm -rf build && node scripts/build.js",
-    "test": "node scripts/test.js",
+    "lint": "eslint src &&  tsc --project ./tsconfig.json --noEmit",
```
Comment (nkshah2) [inline package.json:99]: Refer to auth-react or other js frontend libraries for all the scripts we use for checking if the code is linted and pretty (also make changes to the pre commit for this)

File: .husky/pre-commit [hunk 6]
```diff
@@ -0,0 +1,4 @@
...
+. "$(dirname -- "$0")/_/husky.sh"
+
+npx pretty-quick --staged
```
Comment (nkshah2) [inline .husky/pre-commit:4]: Why is this file still here?

**Review by nkshah2 — CHANGES_REQUESTED** [review]

File: package.json (hunk 5 above)
Comment (sulha199) [reply package.json]: pre-commit is updated

File: .husky/pre-commit (hunk 6 above)
Comment (sulha199) [reply .husky/pre-commit]: removed

File: hooks/pre-commit [hunk 7]
```diff
@@ -26,6 +26,34 @@ else
    echo "$(tput setaf 1)* No$(tput sgr 0)"
 fi
 
+npm run pretty-check >/dev/null 2>/dev/null
```
Comment (nkshah2) [inline hooks/pre-commit:29]: Other SDKs also have a way of checking for circular dependencies, we should add that here

**Review by nkshah2 — CHANGES_REQUESTED** [review]

File: hooks/pre-commit (hunk 7 above)
Comment (sulha199) [reply hooks/pre-commit:41]: its added 
image

**Review by nkshah2 — APPROVED** [review]

**Pull request merged**